### DISCUSSION TITLE
Build errors

### DISCUSSION THREAD

**Discussion by trigaten** [microsoft/psi#159] [Q&A] [answered]
Hello,

I am trying to build \psi from source (on Windows 10 Pro) and am running into a few errors. I have followed your instructions in the Wiki, but get errors like:

CS006 'C:\Users\psi\source\repos\psi3\Sources\Media\Microsoft.Psi.Media_Interop.Windows.x64\x64\Release\Microsoft.Psi.Media_Interop.Windows.x64.dll' could not be found	

and

MSB8040	Spectre-mitigated libraries are required for this project. Install them from the Visual Studio installer (Individual components tab) for any toolsets and architectures being used. Learn more: https://aka.ms/Ofhn4c	Microsoft.Psi.RealSense_Interop.Windows.x64

I have checked that I have the Spectre-mitigated libs installed as well as the C++ ATL with Spectre Mitigations.

Why might this be getting these errors?

**Comment by AshleyF** [comment, marked as answer]
Hi, I can repro this build error as well, but only if I uninstall the Spectre libs. Can you try launching the Visual Studio Installer and under Individual Components filter to `Spectre x64 latest`. Without this filter there are *many* flavors to select. Be sure to select `x86`/`x64` and `latest` (v142) versions. Hopefully, a simple version or platform mismatch is the issue.

![Spectre](https://user-images.githubusercontent.com/692936/121075929-a144ff80-c78a-11eb-9cd4-e5722bfb53ab.png)

**Reply by trigaten** [reply]
![Capture](https://user-images.githubusercontent.com/42948753/121232886-06b6f000-c860-11eb-85d0-69446cb839b1.PNG)
Thank you-- That fixes the Spectre Mitigated libs issues. It seems it was just a version mismatch. I was unsure whether the following install instruction in the Wiki meant to install the latest (v14.xx) version or the version that said (latest version). Thanks for clearing this up.

MSVC v142 - VS 2019 C++ x64/x86 Spectre-mitigated libs (v14.xx, latest version)

Unfortunately, the RealSense is still not building. I installed the SDK from here: https://github.com/IntelRealSense/librealsense/releases/download/v2.45.0/Intel.RealSense.SDK-WIN10-2.45.0.3212.exe

And followed the exact environment variable instruction.

I am getting the errors in the attached image.

**Reply by AshleyF** [reply]
I installed the SDK version you link to above and cannot seem to repro. I can force a repro of this same error (missing include file librealsense2/rs.hpp) by changing my `RealSenseSDKDir` environment variable to some random path. So, I'm thinking that your path may be incorrect? For me, with defaults during installation, it's `C:\Program Files (x86)\Intel RealSense SDK 2.0`. You could verify that `include\librealsense2\rs.hpp` is in there as well. I also noticed that changes to this environment variable are not picked up by Visual Studio until a re-launch of VS. Perhaps that could be the problem as well.

**Reply by trigaten** [reply]
re-launching seems to have resolved all errors.
